operators/constellation-node-operator/internal/infrastructure/scalinggroup_controller.go

```diff
@@ -0,0 +1,137 @@
+type ExternalScalingGroupReconciler struct {
+	// uid is the unique identifier of the Constellation cluster.
+	uid                    string
+	scalingGroupDiscoverer scalingGroupDiscoverer
+	k8sClient              k8sReadWriter
+}
+
+// NewExternalScalingGroupReconciler creates a new InfrastructureReconciler.
+func NewExternalScalingGroupReconciler(uid string, discoverer scalingGroupDiscoverer, k8sClient k8sReadWriter) *ExternalScalingGroupReconciler {
+	return &ExternalScalingGroupReconciler{
+		uid:                    uid,
+		scalingGroupDiscoverer: discoverer,
+		k8sClient:              k8sClient,
+	}
+}
+
+// Reconcile reconciles on scaling groups in CSP infrastructure.
+func (r *ExternalScalingGroupReconciler) Reconcile(ctx context.Context) (executor.Result, error) {
+	logr := log.FromContext(ctx)
+	logr.Info("reconciling external scaling groups")
+
+	nodeGroups, err := r.scalingGroupDiscoverer.ListScalingGroups(ctx, r.uid)
+	if err != nil {
+		return executor.Result{}, err
+	}
+
+	existingNodeGroups := map[string]struct{}{}
+
+	// create all scaling groups that are newly discovered
+	for _, group := range nodeGroups {
+		err := createScalingGroupIfNotExists(ctx, newScalingGroupConfig{
+			k8sClient:            r.k8sClient,
+			resourceName:         group.Name,
+			groupID:              group.GroupID,
+			nodeGroupName:        group.NodeGroupName,
+			autoscalingGroupName: group.AutoscalingGroupName,
+			role:                 group.Role,
+		})
+		if err != nil {
+			return executor.Result{}, err
+		}
+		existingNodeGroups[group.Name] = struct{}{}
+	}
+
+	logr.Info("ensured scaling groups are created", "count", len(nodeGroups))
+
+	// delete all scaling groups that no longer exist
+	var scalingGroups updatev1alpha1.ScalingGroupList
+	if err := r.k8sClient.List(ctx, &scalingGroups); err != nil {
+		return executor.Result{}, err
+	}
+	for _, group := range scalingGroups.Items {
+		if _, ok := existingNodeGroups[group.Name]; !ok {
+			logr.Info("deleting scaling group", "name", group.Name)
+			err := r.k8sClient.Delete(ctx, &group)
+			if err != nil {
+				return executor.Result{}, err
+			}
+		}
+	}
+
+	logr.Info("external scaling groups reconciled")
+
+	return executor.Result{}, nil
+}
+
+func createScalingGroupIfNotExists(ctx context.Context, config newScalingGroupConfig) error {
+	logr := log.FromContext(ctx)
+	err := config.k8sClient.Create(ctx, &updatev1alpha1.ScalingGroup{
+		TypeMeta: metav1.TypeMeta{APIVersion: "update.edgeless.systems/v1alpha1", Kind: "ScalingGroup"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: strings.ToLower(config.resourceName),
+		},
+		Spec: updatev1alpha1.ScalingGroupSpec{
+			NodeVersion:         mainconstants.NodeVersionResourceName,
+			GroupID:             config.groupID,
+			AutoscalerGroupName: config.autoscalingGroupName,
+			NodeGroupName:       config.nodeGroupName,
+			Min:                 1,
+			Max:                 10,
```

Comment:
should this be configurable / use constant?

Reply:
> should this be configurable

This is just a sensible default with some minimun, some maximum and autoscaling disabled (false).
The docs explain how users can configure this after the resource is created.

> use constant

Sounds good